app/src/test/java/org/vss/AbstractKVStoreIntegrationTest.java

```diff
@@ -163,6 +179,177 @@ void getShouldReturnCorrectValueWhenKeyExists() {
     assertThat(response.getValue().toStringUtf8(), is("k3v1"));
   }
 
+  @Test
+  void listShouldReturnPaginatedResponse() {
+
+    int totalKvObjects = 1000;
+    for (int i = 0; i < totalKvObjects; i++) {
+      putObjects((long) i, List.of(kv("k" + i, "k1v1", 0)));
+    }
+    // Overwrite k1 once and k2 twice.
+    putObjects(1000L, List.of(kv("k1", "k1v2", 1)));
+    putObjects(1001L, List.of(kv("k2", "k2v2", 1)));
+    putObjects(1002L, List.of(kv("k2", "k2v3", 2)));
+
+    ListKeyVersionsResponse previousPage = null;
+    List<KeyValue> allKeyVersions = new ArrayList<>();
+
+    while (previousPage == null || !previousPage.getKeyVersionsList().isEmpty()) {
+      ListKeyVersionsResponse currentPage;
+
+      if (previousPage == null) {
+        currentPage = list(null, null, null);
```

Comment:
I'd find this more readable if `list` took a builder.

Reply:
listRequest follows a builder pattern, but this particular "list" function is just for testcase to avoid making them too verbose. Whenever we are using any argument, we extract it to variable and use it for readability.

Reply:
i earlier had that builder everywhere, seems more readable like this 😅

Reply:
Yeah, feel free to keep as is. Could possibly be a little more readable if there were special case functions that called `list` to avoid all the `null`s. But I'll leave that decision to you.